src/cata_tiles.cpp

```diff
@@ -1228,13 +1327,25 @@ bool cata_tiles::draw_sprite_at(const std::vector<int> &spritelist, int x, int y
     destination.w = tile_width;
     destination.h = tile_height;
 
+    if ( svlist.empty() ) {
+        // render nothing
+        return true;
+    }
+
+    const std::vector<int> *spritelist = svlist.pick(loc_rand);
```

Comment:
I'm not a fan of using auto so much when it doesn't significantly improve readability (such as in range loops). As I've mentioned in our previous discussions, it makes it very hard for someone new to the codebase to figure out the type of things and follow the code.

The at() calls were because I couldn't figure out the right syntax to use operator[] on a pointer to a vector. Switching to a reference fixes that, thanks.